File: src/pooh3/mobi/app/a01_clock/model/ClockFactory.java
package pooh3.mobi.app.a01_clock.model;

import static pooh3.mobi.Preconditions.checkArgument;
import static pooh3.mobi.Preconditions.checkNotNull;

public class ClockFactory {

    private static final String DIV = ":";
    private static final int HOUR_INDEX = 0;
    private static final int MINUTE_INDEX = 1;


    public static Clock createByStr(String hourAndMinute) {
        String[] pair =
                checkNotNull(hourAndMinute,
                        "hourAndMinute require not null.").split(DIV);

        checkArgument(pair.length == 2,
                "check format:" + hourAndMinute);

        return new Clock(
                new ShortHand(Integer.parseInt(pair[HOUR_INDEX])),
                new LongHand(Integer.parseInt(pair[MINUTE_INDEX]))
        );
    }
}

File: src/pooh3/mobi/app/a01_clock/model/ClockFactory.kt
package pooh3.mobi.app.a01_clock.model

import pooh3.mobi.elseThrow
import pooh3.mobi.checkNotNull

object ClockFactory {

    private val DIV = ":"
    private val HOUR_INDEX = 0
    private val MINUTE_INDEX = 1

    fun createByStr(hourAndMinute: String): Clock {
        hourAndMinute
                .checkNotNull("hourAndMinute require not null.")
                .split(DIV).dropLastWhile(String::isEmpty).toTypedArray()
                .let {
                    (it.size == 2).elseThrow(
                            "check format:$hourAndMinute  \n eg. \"10:10\"")

                    return Clock.build(
                            it[HOUR_INDEX].parseInt.shortHand,
                            it[MINUTE_INDEX].parseInt.longHand
                    )
                }
    }
}

val String.parseInt: Int
    get() = Integer.parseInt(this)
